File: Cryptographic_Frontend/src/methods/SignInitMethod.cpp
#include "methods/SignInitMethod.hpp"
#include "ResponseMessage.hpp"

#include "StringArgument.hpp"
#include "IntegerArgument.hpp"

cf::SignInitMethod::SignInitMethod(std::string mechanism, int privateKeyHandler)
    : Method("SignInit")
{
    addArgument(ArgumentPtr(new StringArgument("mechanism", mechanism)));
    addArgument(ArgumentPtr(new IntegerArgument("handler", privateKeyHandler)));
}


cf::ResponseMessageStrategy cf::SignInitMethod::getStrategy() const
{
    return [](std::string const & message) -> ResponseMessagePtr {
        ResponseMessagePtr response(new ResponseMessage());
        return std::move(response);
    };
}
// kate: indent-mode cstyle; indent-width 4; replace-tabs on; 

File: Cryptographic_Backend/src/main/java/cb/backend/methods/SignInitMethod.java
package cb.backend.methods;

import cb.backend.*;

public abstract class SignInitMethod implements Method {
	public static class Args {
		public final String mechanism;
		public final long handler;

        public Args(String mechanism, long handler) {
            this.mechanism = mechanism;
            this.handler = handler;
        }
    }
	protected final String mechanism;
	protected final long privateKeyHandler;
	
	public SignInitMethod(Args args) {
		this.mechanism = args.mechanism;
		this.privateKeyHandler = args.handler;
	}
}
